src/python/pants/core/goals/tailor.py

```diff
@@ -297,6 +302,18 @@ def alias_for(self, standard_type: str) -> str | None:
         # This cast suffices to avoid typecheck errors.
         return cast(str, self.options.alias_mapping.get(standard_type))
 
+    def validate_build_file_name(self, build_file_options: BuildFileOptions) -> None:
+        """Check that the specified BUILD file name works with the repository's BUILD file
+        patterns."""
+        if not build_file_options.matches(self.build_file_name):
+            raise ValueError(
+                f"The option `[{self.options_scope}].build_file_name` is set to "
+                f"`{self.build_file_name}`, which is not compatible with "
+                f"`[GLOBAL].build_patterns` ({sorted(build_file_options.patterns)}) "
+                f"and `[GLOBAL].build_ignore` ({sorted(build_file_options.ignores)}).\n\n"
+                "To fix, please update these three options so that they are compatible."
```

Comment:
Maybe rephase to make clear that the written BUILD files will be ignored because of the incompatibility? This will identify to the user why the incompatibility needs to be fixed.

Reply:
Sure. Do you agree with erroring, or think this should be an `.error()` log? My concern with log is that it's easy to miss because the goal's output can be long.

Reply:
It should error. Pants is not going to do what the users expects. They expect `tailor` to write operable BUILD files, but `tailor` instead will be writing ineffective BUILD files. The user wont realize this fact until later. We should inform them at this moment when Pants knows that the written files will be a no-op.